lib/stemcell/launcher.rb

```diff
@@ -203,17 +202,19 @@ def find_instance(id)
       return @ec2.instances[id]
     end
 
-    def kill(instances,opts={})
-      return if instances.nil?
-      instances.each do |i|
+    def kill(instance_ids, opts={})
+      return if instances_ids.nil?
+
+      errors = run_batch_operation(instance_ids) do |id|
         begin
-          instance = find_instance(i)
+          instance = find_instance(id)
           @log.warn "Terminating instance #{instance.instance_id}"
           instance.terminate
         rescue AWS::EC2::Errors::InvalidInstanceID::NotFound => e
-          throw e unless opts[:ignore_not_found]
+          opts[:ignore_not_found] ? nil : e
         end
       end
+      check_errors(:kill, instance, errors)
```

Comment:
That's so right!! I'd have paid more attention when writing production code with dynamic programming language.
I've set up the `ruby-lint` and to capture such kind of errors (and there are several more of such errors). I will also add tests for `kill()` and `set_tags()`.